agent/engine/docker_task_engine.go

```diff
@@ -586,10 +603,15 @@ func (engine *DockerTaskEngine) serialPull(task *api.Task, container *api.Contai
 	defer seelog.Debugf("Released ImagePullDeleteLock after pulling image - %s. Task: %v", container.Image, task)
 	defer ImagePullDeleteLock.Unlock()
 
-	pullStart := time.Now()
+	pullStart := engine.time().Now()
 	defer func(startTime time.Time) {
 		seelog.Infof("Finished pulling container %v in %s. Task: %v", container.Image, time.Since(startTime).String(), task)
 	}(pullStart)
+	ok := task.SetPullStartedAt(pullStart)
+	if ok {
+		seelog.Infof("Task started pull, task %s, time: %s", task.String(), pullStart)
```

Comment:
Here too. Let's be more explicit about recording a timestamp here:  `"Recording timestamp for starting image pull, task: ..."`